src/main/java/org/apache/commons/cli/help/HelpFormatter.java

```diff
@@ -0,0 +1,286 @@
+         */
+        public Builder setOptionGroupSeparator(final String optionGroupSeparator) {
+            this.optionGroupSeparator = Util.defaultValue(optionGroupSeparator, "");
+            return this;
+        }
+
+        /**
+         * Sets the comparator to use for sorting opitons.  If set to {@code null} no sorting is performed.
+         * @param comparator The comparator to use for sorting opitons.
+         * @return this
+         */
+        public Builder setComparator(final Comparator<Option> comparator) {
+            this.comparator = comparator;
+            return this;
+        }
+
+        /**
+         * performs a sanity check and sets default values if they are not set.
+         * @return this.
+         */
+        private Builder sanityCheck() {
+            if (helpWriter == null) {
+                helpWriter = new TextHelpWriter(System.out);
+            }
+            if (optionFormatBuilder == null) {
+                optionFormatBuilder = new OptionFormatter.Builder();
+            }
+            return this;
+        }
+
+        /**
+         * Constructs the {@link HelpFormatter}.
+         * @return this.
+         */
+        public HelpFormatter build() {
+            sanityCheck();
+            return new HelpFormatter(this);
+        }
+    }
+
+    /**
+     * Constructs a new instance using the default {@link Builder}.
+     * @see Builder
+     */
+    public HelpFormatter() {
+        this(new Builder().sanityCheck());
+    }
+
+    /**
+     * Convenience constructor to create an instance using the specified {@link HelpWriter} and the
+     * remaining default {@link Builder}.
+     * @param helpWriter the {@link HelpWriter} to use.
+     */
+    public HelpFormatter(final HelpWriter helpWriter) {
+        this(new Builder().setSerializer(helpWriter).sanityCheck());
+    }
+
+    /**
+     * Constructs the Help formatter.
+     * @param builder the Builder to build from.
+     */
+    private HelpFormatter(final Builder builder) {
+        super(builder.helpWriter, builder.optionFormatBuilder, builder.defaultTableBuilder, builder.comparator,
+                builder.optionGroupSeparator);
+
+        this.showSince = builder.showSince;
+        if (this.tableDefBuilder == null) {
+            tableDefBuilder = this::defaultTableBuilder;
+        }
+    }
+
+    /**
+     * Sets the state of the showSince flag.
+     * @param showSince the desires state of the showSince flag.
+     */
+    public void setShowSince(final boolean showSince) {
+        this.showSince = showSince;
+    }
+
+    /**
+     * The default table builder for the HelpFormatter.  If a different formatter is not specified in the
+     * {@link Builder} this method is used.
+     * @param options the collection of {@link Option} instances to create the table from.
+     * @return A {@link TableDef} to display the options.
+     */
+    public TableDef defaultTableBuilder(final Iterable<Option> options) {
```

Comment:
This is a bit odd. If it's the default behavior AND it's overridable from the builder, then it should be _the_ default value. This would also help validate that the class is configurable enough if the default behavior is coded the same way a user would override it. IOW, can I call the Build with the behavior in this method?